collections/_posts/2019-08-22-self-hosting-a-shadowsocks-vpn-with-outline.md

```diff
@@ -0,0 +1,84 @@
+
+Another factor to consider is your provider's network and latency. Choosing a server closer to you (geographically speaking) will give you better latency, and choosing a server with good bandwidth (>1 Gbps) will minimize the performance hit when using the VPN. Both factors are important to keeping a good browsing experience, but keep in mind using *any* VPN will always be slower than just your plain old internet connection.
+
+Finally, if you want to go with DigitalOcean you can use my affiliate link to receive a $50 credit: [https://m.do.co/c/fb6730f5bb99](https://m.do.co/c/fb6730f5bb99). That's 10 months of free VPN hosting, at $5/month/server. Don't feel obligated to use this link, but you'll receive free credit, and if you spend $25 with DigitalOcean after using it I will get credited, which will enable me to continue writing guides like this! DigitalOcean has a great performing network in my personal experience, and in the experience of the Outline team it works well in regard to circumventing censorship: Not many IP addresses of theirs are blocked.
+
+For this guide we are not going to use an automatic provider in Outline Manager, rather we will manually configure a Linux server. We are using Debian 10. Other distros may work as well, but you may need to install Docker manually.
+
+### Step 3 — Configure Your Server
+
+First, we need to install `curl`. Connect to your server via SSH and enter the following commands:
+
+```
+sudo apt update
+sudo apt install curl
+```
+
+Next open Outline Manager on your local machine and you should be given 4 options to configure a server. Select the "Set Up" button under the "Advanced, Set up Outline anywhere" option.
+
+![](/assets/img/2019-08-22-outline-server-setup.png){: .w-100}
+
+Outline will give you a string to paste. Connect to your server over SSH and paste the code in the Outline Manager box into the Terminal. The process will take a minute or two and will ask you a couple questions. You can just press enter to accept the default configuration whenever it asks.
+
+After it completes, it will give you a long line starting with `{"apiUrl"` (depending on your Terminal's color support it will appear as green). Copy that line, and paste it in the second box back in Outline Manager. Then, click "Done".
+
+### Step 4 — Connect Your Devices
+
+Download the Outline app on the device you want to connect. Outline has applications for the following operating systems:
+
+- [Android](https://play.google.com/store/apps/details?id=org.outline.android.client)
```

Comment:
Considering who we are, is there F-Droid or similar available that you could mention?

Reply:
Usually it seems like F-Droid and other Play alternatives hijack Play Store links so it would still work in theory, although I don’t know if it is actually on F-Droid. I believe I linked to GitHub where the builds are available as well though.